webapp/sources/rudder/rudder-web/src/main/elm/directivecompliance/sources/ApiCalls.elm

```diff
@@ -0,0 +1,95 @@
+module ApiCalls exposing (..)
+
+import DataTypes exposing (..)
+import Http exposing (..)
+import JsonDecoder exposing (..)
+import Url.Builder exposing (QueryParameter)
+
+
+--
+-- This files contains all API calls for the Directive compliance UI
+--
+
+getUrl: DataTypes.Model -> List String -> List QueryParameter -> String
+getUrl m url p=
+  Url.Builder.relative (m.contextPath :: "secure" :: "api"  :: url) p
+
+getPolicyMode : Model -> Cmd Msg
+getPolicyMode model =
+  let
+    req =
+      request
+        { method  = "GET"
+        , headers = []
+        , url     = getUrl model [ "settings", "global_policy_mode" ] []
+        , body    = emptyBody
+        , expect  = expectJson GetPolicyModeResult decodeGetPolicyMode
+        , timeout = Nothing
+        , tracker = Nothing
+        }
+  in
+    req
+
+getAllRules : Model -> Cmd Msg
+getAllRules model =
+  let
+    req =
+      request
+        { method  = "GET"
+        , headers = []
+        , url     = getUrl model [ "rules" ] []
+        , body    = emptyBody
+        , expect  = expectJson GetRulesList decodeGetRules
+        , timeout = Nothing
+        , tracker = Nothing
+        }
+  in
+    req
+
+getNodesList : Model -> Cmd Msg
+getNodesList model =
+  let
+    req =
+      request
+        { method  = "GET"
+        , headers = []
+        , url     = getUrl model ["nodes"] []
```

Comment:
i'm unsure as to why you need that. If it's really necessary, you could fetfh nly the minimal version of the node, or minimal+ exactly what you need